src/components/Insights/InsightsCatalog/index.tsx

```diff
@@ -132,59 +185,96 @@ export const InsightsCatalog = (props: InsightsCatalogProps) => {
-          onChange={(val: Sorting) => {
-            setSorting(val);
-          }}
-          options={[
-            ...(defaultQuery.insightViewType === "Issues"
-              ? [
-                  {
-                    value: SORTING_CRITERION.CRITICAL_INSIGHTS,
-                    label: "Critical issues",
-                    defaultOrder: SORTING_ORDER.DESC
-                  }
-                ]
-              : []),
-            {
-              value: SORTING_CRITERION.LATEST,
-              label: "Latest",
-              defaultOrder: SORTING_ORDER.DESC
-            }
-          ]}
-          default={defaultQuery.sorting}
-        />
-        <Tooltip title={"Refresh"}>
-          <s.RefreshButton
-            buttonType={"tertiary"}
-            icon={RefreshIcon}
-            onClick={handleRefreshButtonClick}
+        <s.ToolbarRow>
+          <SearchInput
+            disabled={Boolean(config.scope?.span)}
+            onChange={(val: string | null) => {
+              setSearchInputValue(val);
+            }}
+            value={searchInputValue}
           />
-        </Tooltip>
-      </s.Toolbar>
-      {mode === ViewMode.OnlyDismissed && (
-        <s.InsightsViewModeToolbar>
-          <Button
-            buttonType={"tertiary"}
-            label={"Back to All Insights"}
-            icon={(props) => (
-              <ChevronIcon {...props} direction={Direction.LEFT} />
-            )}
-            onClick={() => setMode(ViewMode.All)}
+          <SortingSelector
+            onChange={(val: Sorting) => {
+              setSorting(val);
+            }}
+            options={[
+              ...(defaultQuery.insightViewType === "Issues"
+                ? [
+                    {
+                      value: SORTING_CRITERION.CRITICAL_INSIGHTS,
+                      label: "Critical issues",
+                      defaultOrder: SORTING_ORDER.DESC
+                    }
+                  ]
+                : []),
+              {
+                value: SORTING_CRITERION.LATEST,
+                label: "Latest",
+                defaultOrder: SORTING_ORDER.DESC
+              }
+            ]}
+            default={defaultQuery.sorting}
           />
-          <s.DismissedDescription>
-            <s.DismissedCount>{insights.length}</s.DismissedCount>
-            dismissed insights
-          </s.DismissedDescription>
-        </s.InsightsViewModeToolbar>
-      )}
+          <Tooltip title={"Refresh"}>
+            <s.RefreshButton
+              buttonType={"tertiary"}
+              icon={RefreshIcon}
+              onClick={handleRefreshButtonClick}
+            />
+          </Tooltip>
+        </s.ToolbarRow>
+        {mode === ViewMode.All ? (
+          isMarkingAsReadToolbarVisible && (
+            <s.ViewModeToolbarRow>
+              <s.InsightCountDescription>
+                <s.InsightCount>{props.unreadCount}</s.InsightCount>
+                unread issue{props.unreadCount === 1 ? "" : "s"}
```

Comment:
can we use for this humanizer package?

Reply:
I added the `formatUnit` utility for now